plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt

```diff
@@ -354,43 +362,64 @@ open class KtlintPlugin : Plugin<Project> {
         reporters.set(extension.reporters)
     }
 
-    private fun Project.getMetaKtlintCheckTask(): Task = this.tasks.findByName(CHECK_PARENT_TASK_NAME)
-        ?: this.task(CHECK_PARENT_TASK_NAME).apply {
+    private fun Project.getMetaKtlintCheckTask(): TaskProvider<Task> = try {
+        this.tasks.named(CHECK_PARENT_TASK_NAME)
+    } catch (ignored: UnknownDomainObjectException) {
+        taskHelper(CHECK_PARENT_TASK_NAME) {
             group = VERIFICATION_GROUP
             description = "Runs ktlint on all kotlin sources in this project."
         }
+    }
 
     private fun Project.getAndroidVariantMetaKtlintCheckTask(
         variantName: String,
         multiplatformTargetName: String? = null
-    ): Task {
+    ): TaskProvider<Task> {
         val taskName = "ktlint${variantName.capitalize()}${multiplatformTargetName?.capitalize() ?: ""}Check"
-        return tasks.maybeCreate(taskName).apply {
```

Comment:
I see. Maybe create is not available anymore, and they didn't provide any lazy alternative, it seems.